src/NUnitFramework/tests/Internal/Filters/AndFilterTests.cs

```diff
@@ -71,6 +71,74 @@ public void ExplicitMatchTest()
+            Assert.False(filter.IsExplicitMatch(_dummyFixture.Tests[0]));
+
+            Assert.That(filter.Match(_anotherFixture), "4");
+        }
+
+        [TestCase(new bool[] { false, false}, false, StrictIdFilterForTests.EqualValueFunction.IsExplicitMatch)]
+        [TestCase(new bool[] { true, false }, false, StrictIdFilterForTests.EqualValueFunction.IsExplicitMatch)]
+        [TestCase(new bool[] { false, true }, false, StrictIdFilterForTests.EqualValueFunction.IsExplicitMatch)]
+        [TestCase(new bool[] { true, true }, true, StrictIdFilterForTests.EqualValueFunction.IsExplicitMatch)]
+        [TestCase(new bool[] { false, false }, false, StrictIdFilterForTests.EqualValueFunction.Match)]
+        [TestCase(new bool[] { true, false }, false, StrictIdFilterForTests.EqualValueFunction.Match)]
+        [TestCase(new bool[] { false, true }, false, StrictIdFilterForTests.EqualValueFunction.Match)]
+        [TestCase(new bool[] { true, true }, true, StrictIdFilterForTests.EqualValueFunction.Match)]
+        [TestCase(new bool[] { false, false }, false, StrictIdFilterForTests.EqualValueFunction.Pass)]
+        [TestCase(new bool[] { true, false }, false, StrictIdFilterForTests.EqualValueFunction.Pass)]
+        [TestCase(new bool[] { false, true }, false, StrictIdFilterForTests.EqualValueFunction.Pass)]
+        [TestCase(new bool[] { true, true }, true, StrictIdFilterForTests.EqualValueFunction.Pass)]
+        public void CombineTest(IEnumerable<bool> inputBooleans, bool expectedResult,
+            StrictIdFilterForTests.EqualValueFunction expectedCall)
+        {
+            var filters = new List<StrictIdFilterForTests>();
+            foreach (var inputBool in inputBooleans)
+            {
+                string id = string.Empty;
+                if (inputBool)
+                {
+                    id = _dummyFixture.Id;
+                }
+
+                var strictFilter = new StrictIdFilterForTests(id, expectedCall);
+                Assert.AreEqual(inputBool, Match(strictFilter, expectedCall));
+
+                filters.Add(strictFilter);
+            }
+
+            var filter = new AndFilter(filters.ToArray());
+            bool calculatedResult = Match(filter, expectedCall);
+            Assert.AreEqual(expectedResult, calculatedResult);
+        }
+
+        private bool Match(
+            TestFilter filter, StrictIdFilterForTests.EqualValueFunction expectedCall)
+        {
+            switch (expectedCall)
+            {
+                case StrictIdFilterForTests.EqualValueFunction.IsExplicitMatch:
+                    return filter.IsExplicitMatch(_dummyFixture);
+                case StrictIdFilterForTests.EqualValueFunction.Match:
+                    return filter.Match(_dummyFixture);
+                case StrictIdFilterForTests.EqualValueFunction.Pass:
+                    return filter.Pass(_dummyFixture);
+                default:
+                    throw new AssertionException(
```

Comment:
Don't throw the AssertionException, rather, use Assert.Fail as we tell users to do. :-) This will stop working as soon as we change how failures are reported. Alternatively, if you want to convey that this is an error in how the test was written rather than a true failure, throw something else, like an ArgumentException.